dev-qt/qt-creator/qt-creator-9999.ebuild

```diff
@@ -110,7 +110,20 @@ unset x
 PATCHES=( "${FILESDIR}"/${PN}-4.12.0-dylib-fix.patch )
 
 pkg_setup() {
-	use clang && llvm_pkg_setup
+	if use clang; then
```

Comment:
OK.  It might make sense to at least emit an `ewarn` though, seeing as formatting with clang-format is a feature one might expect in a modern IDE with `USE=clang`.  I guess I should formally submit a bug-report and add a GIT commit tag to the sys-devel/clang commit.

The 6 possibilities as I see it:
1. @gentoo/llvm could unconditionally patch clang.
2. @gentoo/llvm could use a `qtcreator-compat` USE flag to conditionally patch clang.
3. qt-creator could be built with its own bundled llvm/clang source from QT repo (which in the `release_100-based` branch differs from vanilla by only two commits).  This would be in line with upstream's intentions but violates Gentoo's philosophy.
4. Check and `ewarn` about unpatched clang but continue with the build.
5. Don't check or `ewarn` and, at runtime, let qt-creator's popup inform the user that the plugin has been disabled and they need to build against patched clang.  This is an option for those who like closing invalid bug reports on gentoo-bugs.
6.  Disable building the clang-format/beautifier plugins altogether.  Not sure I'd want to use an IDE that didn't have this functionality.

Reply:
Dropped for now.

Reply:
After some preliminary testing, it seems that linking the _libClangFormat.so_ plugin with a patched _libclangFormat.a_ clang library works as a solution.  Since _libclangFormat.a_, itself, links to other llvm/clang libs, getting this to work in portage would likely require the build of _libclangFormat.a_ in its own ebuild with its version tied to a clang slot. For instance, _qtcreator-libformat-10.ebuild_.  _dev-qt/qtcreator_ would have a build-time dependency on It with the same slot as `get_llvm_prefix ${LLVM_MAX_SLOT}`.  This would mean three ebuilds for three supported slots.  The ebuilds would be minimal versions of _sys-devel/clang_ (no multilib, no useflag support, etc...).  In `src_compile()`, `cmake-utils_src_compile libclangFormat.a` would build the library.  It, along with _include/clang/Format/Format.h_, could then be copied/installed to a directory that doesn't clash with sys-devel/clang.  Finally, in qt-creator,  src/plugins/clangformat/clangformat.pro could be patched to link against the patched library and build with the patched header.

As an alternative, if you decide to change qt-creator's clang dependence to a single slot, this could probably be done without any extra ebuilds or installations.